src/cls/rbd/cls_rbd.cc

```diff
@@ -6898,17 +6898,71 @@ int dir_remove(cls_method_context_t hctx,
 }
 
 static const string RBD_GROUP_SNAP_KEY_PREFIX = "snapshot_";
+static const string RBD_GROUP_SNAP_ORDER_KEY_PREFIX = "snap_order_";
+static const string RBD_GROUP_SNAP_MAX_ORDER_KEY = "snap_max_order";
 
 std::string snap_key(const std::string &snap_id) {
   ostringstream oss;
   oss << RBD_GROUP_SNAP_KEY_PREFIX << snap_id;
   return oss.str();
 }
 
+std::string snap_order_key(const std::string &snap_id) {
+  ostringstream oss;
+  oss << RBD_GROUP_SNAP_ORDER_KEY_PREFIX << snap_id;
+  return oss.str();
+}
+
+static std::string snap_id_from_order_key(const string &key) {
```

Comment:
While it's better to default to private, it's not done in most of the file including the surrounding methods that are being added, so I'd suggest for consistency:
```suggestion
std::string snap_id_from_order_key(const string &key) {
```